lib/client/redirect.go

```diff
@@ -336,12 +361,36 @@ func (rd *Redirector) wrapCallback(fn func(http.ResponseWriter, *http.Request) (
 			case rd.errorC <- err:
 			case <-rd.context.Done():
 			}
-			http.Redirect(w, r, errorURL, http.StatusFound)
+			redirectURL := errorURL
+			// A second SSO login attempt will be initiated if a key policy requirement was not satisfied.
+			if requiredPolicy, err := keys.ParsePrivateKeyPolicyError(err); err == nil && rd.PrivateKeyPolicy != keys.PrivateKeyPolicyNone {
```

Comment:
Does `rd.PrivateKeyPolicy` get set when the hardware key policy was unknown at time of login attempt? IIUC from my testing rd.PrivateKeyPolicy is always empty here. I think we might need a different field to act as an old-version check